python/src/scipp/plot/view1d.py

```diff
@@ -0,0 +1,137 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @author Neil Vaytet
+
+# Scipp imports
+from .figure1d import PlotFigure1d
+from .._utils import make_random_color
+
+
+class PlotView1d:
```

Comment:
It is not clear to me why we have `PlotView1d` and `PlotFigure1d`? Can you explain?

Reply:
The `figure1d` is basically just the raw mpl figure that can also duplicate or remove lines, while the view adds the additional functionality that handles plotting profiles (sending click events to the controller) as well as adding vertical lines to show where the profiles have been taken.
In the case of making a plot with `projection='1d'` (the 'superplot'), your `view1d` will hold a `figure1d`, but the `profile` will also hold a `figure1d`.
you can save a current slice on the `view1d` (superplot), but you can also save a current profile on the `profileview`. Both are using `figure1d` internally.

In 2d, the `figure2d` displays the image, while the `view2d` will also handle the stuff to do with profiles, such as overlaying the little couloured dots onto the image to show where the profiles were taken.

Reply:
👍 
Maybe explain this a bit in the docstrings of the classes?

Reply:
Good suggestion!